lightning/src/ln/functional_tests.rs

```diff
@@ -257,7 +257,7 @@ fn test_update_fee_unordered_raa() {
 	// ...but before it's delivered, nodes[1] starts to send a payment back to nodes[0]...
 	let (_, our_payment_hash) = get_payment_preimage_hash!(nodes[0]);
 	let net_graph_msg_handler = &nodes[1].net_graph_msg_handler;
-	nodes[1].node.send_payment(&get_route(&nodes[1].node.get_our_node_id(), net_graph_msg_handler, &nodes[0].node.get_our_node_id(), None, &Vec::new(), 40000, TEST_FINAL_CLTV, &logger).unwrap(), our_payment_hash, &None).unwrap();
+	nodes[1].node.send_payment(&get_route(&nodes[1].node.get_our_node_id(), &*net_graph_msg_handler.network_graph.read().unwrap(), &nodes[0].node.get_our_node_id(), None, &Vec::new(), 40000, TEST_FINAL_CLTV, &logger).unwrap(), our_payment_hash, &None).unwrap();
```

Comment:
The dereference is unnecessary here and throughout.

Reply:
The types are different - without the ref+deref its a MutexGuard, with the ref+deref you get the NetworkGraph itself.

Reply:
You shouldn't need to explicitly dereference a `MutexGuard`. See how it is working correctly in `fuzz/src/full_stack.rs` without the dereference.

Reply:
Actually, not sure if that is the same case, but it does compile here if you remove the dereference.

Reply:
Ah, I'd tried previously without either the & or *, which obviously doesn't work, but you're right, works with only the &.